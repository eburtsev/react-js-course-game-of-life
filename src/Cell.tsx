import React, { FC } from "react";

interface CellProps {
    x: number;
    y: number;
    clicked?: boolean;
    onClick: (x: number, y: number) => void;
}

const style = {
    display: "inline-block",
    height: "3em",
    width: "3em",
    border: "#AAAAAA solid 1px",
    color: "#555555",
    margin: "1px",
    verticalAlign: "top",
};

export const Cell: FC<CellProps> = ({ x, y, clicked = false, onClick }) => {
    return (
        <div id={`Cell_${x}_${y}`} style={style} onClick={() => onClick(x || 0, y || 0)}>
            {clicked ? `${x}/${y}` : ""}
        </div>
    );
};
