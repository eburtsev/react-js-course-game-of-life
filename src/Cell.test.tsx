import React from "react";
import { mount } from "enzyme";
import renderer from "react-test-renderer";
import { Cell } from "./Cell";

describe("Cell tests", () => {
    const x = 4;
    const y = 7;
    test("Default cell", () => {
        const tree = renderer.create(<Cell x={x} y={y} clicked={false} onClick={jest.fn()} />).toJSON();
        expect(tree).toMatchSnapshot();
    });
    test("Clicked cell", () => {
        const tree = renderer.create(<Cell x={x} y={y} clicked={true} onClick={jest.fn()} />).toJSON();
        expect(tree).toMatchSnapshot();
    });
    test("Simulate click", () => {
        const clickHandler = jest.fn();
        mount(<Cell x={x} y={y} clicked={false} onClick={clickHandler} />).simulate("click");
        expect(clickHandler).toHaveBeenCalledWith(x, y);
    });
});
