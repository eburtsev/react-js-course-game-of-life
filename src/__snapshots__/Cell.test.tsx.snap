// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cell tests Clicked cell 1`] = `
<div
  id="Cell_4_7"
  onClick={[Function]}
  style={
    Object {
      "border": "#AAAAAA solid 1px",
      "color": "#555555",
      "display": "inline-block",
      "height": "3em",
      "margin": "1px",
      "verticalAlign": "top",
      "width": "3em",
    }
  }
>
  4/7
</div>
`;

exports[`Cell tests Default cell 1`] = `
<div
  id="Cell_4_7"
  onClick={[Function]}
  style={
    Object {
      "border": "#AAAAAA solid 1px",
      "color": "#555555",
      "display": "inline-block",
      "height": "3em",
      "margin": "1px",
      "verticalAlign": "top",
      "width": "3em",
    }
  }
>
  
</div>
`;
